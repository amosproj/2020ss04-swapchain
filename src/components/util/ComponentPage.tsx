import React from "react"

interface Props {
  title?: string
  children: any
  bg?: string
}

export const ComponentPage = (props: Props) => {
  return (
    <div className="container mx-auto">
      <main className={`pb-10 mx-auto ${props.bg} ${props.bg ? "shadow-sm" : ""}  p-4 sm:px-6 lg:px-8`}>
        <h1 className="py-4 text-3xl font-semibold leading-tight text-gray-900">{props.title}</h1>
        {props.children}
      </main>
    </div>
  )
}
