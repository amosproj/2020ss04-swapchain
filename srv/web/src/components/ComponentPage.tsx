import React from "react"
import { PromiseProvider } from "mongoose"

type Props = {
    title?: string
    children: any
}

export const ComponentPage = (props:Props) => {
    return (
        <div className="container py-10 mx-auto mt-10">
          <main className="pb-10 mx-auto bg-white rounded shadow-sm bg-gray-100p-4 sm:px-6 lg:px-8">
          <h1 className="py-4 text-3xl font-bold leading-tight text-gray-900">
               {props.title}</h1>
             {props.children}
          </main>
        </div>
    )
}