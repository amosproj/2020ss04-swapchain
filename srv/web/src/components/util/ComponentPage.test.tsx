import React from 'react';
import {ComponentPage} from './ComponentPage';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<ComponentPage title="title"><h1>Hello World</h1></ComponentPage>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});