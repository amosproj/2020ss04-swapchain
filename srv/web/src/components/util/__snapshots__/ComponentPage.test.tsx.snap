// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container mx-auto"
>
  <main
    className="pb-10 mx-auto undefined   p-4 sm:px-6 lg:px-8"
  >
    <h1
      className="py-4 text-3xl font-bold leading-tight text-gray-900"
    >
      title
    </h1>
    <h1>
      Hello World
    </h1>
  </main>
</div>
`;
